Fatras/include/ActsFatras/Kernel/Simulator.hpp

```diff
@@ -35,26 +35,20 @@ namespace ActsFatras {
 /// Single particle simulator with a fixed propagator and physics list.
 ///
 /// @tparam generator_t random number generator
-/// @tparam continuous_physics_t physics lists for continuous interactions
-/// @tparam pointlike_physics_t physics lists for point-like interactions
-/// @tparam hit_surface_selector_t sensitive hit surfaces selector
+/// @tparam interactions_t interaction list
+/// @tparam hit_surface_selector_t selector for hit surfaces
 /// @tparam decay_t decay module
-template <typename propagator_t, typename continuous_physics_t,
-          typename pointlike_physics_t, typename hit_surface_selector_t,
-          typename decay_t>
+template <typename propagator_t, typename interactions_t,
+          typename hit_surface_selector_t, typename decay_t>
 struct ParticleSimulator {
   /// How and within which geometry to propagate the particle.
   propagator_t propagator;
-  /// Physics list detailing the simulated continuous interactions.
-  /// Will be copied to the per-call interactor.
-  continuous_physics_t continuous;
-  /// Physics list detailing the simulated point-like interactions.
-  /// Will be copied to the per-call interactor.
-  pointlike_physics_t pointlike;
-  /// Where hits are registiered. Will be copied to the per-call interactor.
-  hit_surface_selector_t selectHitSurface;
   /// Decay module.
   decay_t decay;
+  /// Interaction list containing the simulated interactions.
+  interactions_t interactions;
```

Comment:
```suggestion
  interactions_t interactionLits;
```

maybe ?

Reply:
I was thinking about this myself. I think it does not matter here that the interactions type is typically based on `InteractionsList`. For the simulation it only matters that it provides the necessary interface. I would leave it as-is.